sycl/source/detail/queue_impl.hpp

```diff
@@ -525,32 +535,41 @@ class queue_impl {
                     const std::shared_ptr<queue_impl> &SecondaryQueue,
                     const detail::code_location &Loc,
                     const SubmitPostProcessF *PostProcess) {
-    handler Handler(Self, PrimaryQueue, SecondaryQueue, MHostQueue);
-    Handler.saveCodeLoc(Loc);
-    CGF(Handler);
-
-    // Scheduler will later omit events, that are not required to execute tasks.
-    // Host and interop tasks, however, are not submitted to low-level runtimes
-    // and require separate dependency management.
-    const CG::CGTYPE Type = Handler.getType();
     event Event = detail::createSyclObjFromImpl<event>(
         std::make_shared<detail::event_impl>());
-
-    if (PostProcess) {
-      bool IsKernel = Type == CG::Kernel;
-      bool KernelUsesAssert = false;
-
-      if (IsKernel)
-        // Kernel only uses assert if it's non interop one
-        KernelUsesAssert = !(Handler.MKernel && Handler.MKernel->isInterop()) &&
-                           ProgramManager::getInstance().kernelUsesAssert(
-                               Handler.MOSModuleHandle, Handler.MKernelName);
-
-      finalizeHandler(Handler, Type, Event);
-
-      (*PostProcess)(IsKernel, KernelUsesAssert, Event);
-    } else
-      finalizeHandler(Handler, Type, Event);
+    if (auto graphImpl = Self->getCommandGraph(); graphImpl != nullptr) {
+      std::vector<ext::oneapi::experimental::detail::node_ptr> deps;
+      if (auto lastNode = graphImpl->getLastNode(); lastNode != nullptr) {
```

Comment:
Would stick a quick TODO comment in here with the limitation about submission order

Reply:
I've added a comment explaining this!